### PR TITLE
Payments API: client provided payment_id

### PR DESCRIPTION
**Pull request by cristianbica**

### CODE REVIEW

**Review by dcarneiro — APPROVED** [review]
LGTM 👌

**Review by szymon-jez — APPROVED** [review]
I've read the documentation: :heavy_check_mark:.
I assume that this is live already in BE so I will soon try this all out from CB.

File: api-documentation/payments.md [hunk 1]
```diff
@@ -46,6 +46,10 @@ Last name of the payee. Required unless invoice\_id is provided.
 Amount in cents that is going to be paid. Required unless invoice\_id is provided. Can't be smaller than `100`.
 {% endapi-method-parameter %}
 
+{% api-method-parameter name="payment\_id" type="string" required=false %}
+Payment identifier, will be used further to reference this Payment. If not provided ClubCollect will generate an identifier (a hexadecimal string of at least 40 chars). This values must be unique and have between 10 and 50 chars so we advise you to provide an UUID like value. If ClubCollect determines that the value is not unique an error will be returned.
```
Comment (szymon-jez) [inline api-documentation/payments.md:50]: I have just observed that if I provide a payment ID like `5f283622-5e9d-4b43-834f-cb60b2f9426a` (generated using `Ecto.UUID.generate()`) BE accepts it, but it internally converts it silently to `5f283622-5e9d-4b43-834f-cb60b2f9426a` (so it removes the hyphens). Then I'm not able to ask for the status of the payment using the id I have generated.

I think that BE should either not remove the hyphens or return an argument validation error. I would desire the first one as a user.

This should at least be documented, but I think that this is rather a but than a normal behavior.

Comment (szymon-jez) [reply api-documentation/payments.md:50]: I now see that this is not true. BE seems to not accept `payment_id` and I've got confused with just spotting the last of `-` and not comparing the ids. I think I was to fast starting to test this.

Comment (vruizext) [reply api-documentation/payments.md:50]: @szymon-jez AFIK BE isn't performing any validations on the hyphens. The only requirements to accept the payment ID as valid are the length and the uniqueness. See PR

https://github.com/socialpaymentsbv/billing-engine/pull/9932/files#diff-144ca0bba0d243d9a745c38e0e56d1ea4d102fec7b848f4fa2314662f390785dR173

File: api-documentation/payments.md [hunk 2]
```diff
@@ -99,7 +103,7 @@ Payee email address.
 {% endapi-method-parameter %}
 
 {% api-method-parameter name="phone\_number" type="string" required=false %}
-Payee phone number.
+Payee phone number. It is expected to provide the full phone number including international prefix (e.g. +31)
```
Comment (vruizext) [inline api-documentation/payments.md]:
```suggestion
Payee phone number. It is expected that partner provides the phone number in E.164 format, i.e. including international prefix (e.g. +31)
```

Comment (vruizext) [reply api-documentation/payments.md]: I think it's better to be explicit about the E.164 format, to not leave space for misunderstandings about what a "full phone number" is.

**Review by vruizext — APPROVED** [review]
Looks good 👍🏼 

Left a suggestion ref phone number format.

File: api-documentation/payments.md (hunk 1 above)
Comment (szymon-jez) [reply api-documentation/payments.md:50]: Yes, I was to fast with this. I should have first checked if the BE PR got shipped.

**Pull request merged**